services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/CwaApiStructureProvider.java

```diff
@@ -42,10 +42,14 @@
         ignoredValue -> Set.of(distributionServiceConfig.getApi().getVersionV1()),
         Object::toString);
 
-    versionDirectory.addWritableToAll(ignoredValue ->
-        Optional.of(appConfigurationStructureProvider.getAppConfiguration()));
-    versionDirectory.addWritableToAll(ignoredValue ->
-        Optional.of(diagnosisKeysStructureProvider.getDiagnosisKeys()));
+    versionDirectory.addWritableToAll(
+        ignoredValue -> Optional.of(appConfigurationStructureProvider.getAppConfiguration()));
+    versionDirectory.addWritableToAll(
+        ignoredValue -> Optional.ofNullable(appConfigurationStructureProvider.getAppConfigurationV2ForAndroid()));
+    versionDirectory.addWritableToAll(
+        ignoredValue -> Optional.ofNullable(appConfigurationStructureProvider.getAppConfigurationV2ForIos()));
```

Comment:
I don't see, where the different (simpler) path is being used...

Reply:
What you highlighted here is in the context of generating the folder/writable "../version/v1/". We are adding the Writable structures for the v1 folde which are:
*  appConfigurationStructureProvider.getAppConfiguration(). -> this Writable directory has other writables underneath which will get generated (/configuration/country etc). See AppConfigurationDirectory
* appConfigurationStructureProvider.getAppConfigurationV2ForAndroid() (& its IOS counterpart) will provide the archive file directly ( see AppConfigurationV2StructureProvider.getConfigurationArchive()) . It does not generate extra structures underneath .../v1/....